soap/soap11_endpoint.bal

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+# Object for SOAP 1.1 client endpoint.
+#
+# + soap11Client - Http client created to send SOAP 1.1 requests.
+public type Soap11Client client object {
+
+    private http:Client soap11Client;
+
+    public function __init(string url, SoapConfiguration? soapConfig = ()) {
+        self.soap11Client = new(url, config = soapConfig.clientConfig);
```

Comment:
Shouldn't we type check the soapConfig here? What if nill was passed?

Reply:
This is type Checked in HTTP client initialization.